src/variety/slither.js

```diff
@@ -294,13 +393,31 @@
 			"checkdir4BorderLine",
 
 			"checkOneLoop",
-			"checkDeadendLine+"
+			"checkDeadendLine+",
+
+			"checkSheepInWolfOut@swslither"
 		],
 
 		checkdir4BorderLine: function() {
 			this.checkAllCell(function(cell) {
-				return cell.qnum >= 0 && cell.getdir4BorderLine1() !== cell.qnum;
+				return (
+					cell.qnum >= 0 &&
+					cell.qnum <= 4 &&
+					cell.getdir4BorderLine1() !== cell.qnum
+				);
 			}, "nmLineNe");
+		},
+
+		checkSheepInWolfOut: function() {
```

Comment:
I prefer the Sheep check and the Wolf check to be separate, instead of being combined.

Reply:
Yeah ... I had it that way initially but found when solving I preferred the check to highlight all instances of a similar type of error simultaneously.

Happy to change if you really think it's better.

Also, I wasn't sure of the name for these error codes. Is there a prefix you think would be appropriate that's in line with the others?

Reply:
`nmInside` and `nmOutside` would be fine.

Reply:
k, done